src/main/java/com/exadel/ipromise/dao/impl/PromiseDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.exadel.ipromise.dao.impl;
+
+import com.exadel.ipromise.dao.PromiseDao;
+import com.exadel.ipromise.entity.Promise;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+
+@Repository
+public class PromiseDaoImpl implements PromiseDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    final String CREATE = "INSERT INTO jpa.promises (user_id, addiction_id, start_date_stamp, amount_days) VALUES(?, ?, ?, ?)";
+    final String GET_BY_ID_USER = "SELECT * FROM jpa.promises AS PR INNER JOIN jpa.addictions AS ADCT on ADCT.addiction_id = PR.addiction_id WHERE user_id = ?";
+
+    public PromiseDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public Long create(Promise promise) {
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement preparedStatement = connection.prepareStatement(CREATE, new String[]{"promise_id"});
+            preparedStatement.setInt(1, promise.getUserId());
+            preparedStatement.setInt(2, promise.getAddictionId());
+            preparedStatement.setInt(3, promise.getStartDateStamp());
+            preparedStatement.setInt(4, promise.getAmountDays());
+            return preparedStatement;
+        }, keyHolder);
+        return (Long) keyHolder.getKey();
+    }
+
+    public List<Promise> getByIdUser(Long userId) {
```

Comment:
seems you not use it. You User object, which you sent to the client, must contains all user promises.

Reply:
At the moment I made only the user without promises. This was done purposefully, so as not to complicate the PR. When I create a controller and a "promise" service, it will not be difficult for me to add "promises" to the User